ingestion/src/metadata/ingestion/source/database/stored_procedures_mixin.py

```diff
@@ -0,0 +1,182 @@
+            except Exception as exc:
+                self.status.failed(
+                    StackTraceError(
+                        name="Stored Procedure",
+                        error=f"Error trying to get procedure name due to [{exc}]",
+                        stack_trace=traceback.format_exc(),
+                    )
+                )
+
+        return queries_dict
+
+    @staticmethod
+    def is_lineage_query(query_type: str, query_text: str) -> bool:
+        """Check if it's worth it to parse the query for lineage"""
+
+        if query_type in ("MERGE", "UPDATE", "CREATE_TABLE_AS_SELECT"):
+            return True
+
+        if query_type == "INSERT" and re.search(
+            "^.*insert.*into.*select.*$", query_text, re.IGNORECASE
+        ):
+            return True
+
+        return False
+
+    def yield_procedure_lineage(
+        self, query_by_procedure: QueryByProcedure
+    ) -> Iterable[Either[AddLineageRequest]]:
+        """Add procedure lineage from its query"""
+
+        self.update_context(key="stored_procedure_query_lineage", value=False)
+        if self.is_lineage_query(
+            query_type=query_by_procedure.query_type,
+            query_text=query_by_procedure.query_text,
+        ):
+            self.update_context(key="stored_procedure_query_lineage", value=True)
+
+            for either_lineage in get_lineage_by_query(
+                self.metadata,
+                query=query_by_procedure.query_text,
+                service_name=self.context.database_service.name.__root__,
+                database_name=self.context.database.name.__root__,
+                schema_name=self.context.database_schema.name.__root__,
+                dialect=ConnectionTypeDialectMapper.dialect_of(
+                    self.context.database_service.serviceType.value
+                ),
+                timeout_seconds=self.source_config.queryParsingTimeoutLimit,
+                lineage_source=LineageSource.QueryLineage,
+            ):
+                if either_lineage.right.edge.lineageDetails:
+                    either_lineage.right.edge.lineageDetails.pipeline = EntityReference(
+                        id=self.context.stored_procedure.id,
+                        type="storedProcedure",
+                    )
+
+                yield either_lineage
+
+    def yield_procedure_query(
+        self, query_by_procedure: QueryByProcedure
+    ) -> Iterable[Either[CreateQueryRequest]]:
+        """Check the queries triggered by the procedure and add their lineage, if any"""
+
+        yield Either(
+            right=CreateQueryRequest(
+                query=SqlQuery(__root__=query_by_procedure.query_text),
+                query_type=query_by_procedure.query_type,
+                duration=query_by_procedure.query_duration,
+                queryDate=Timestamp(
+                    __root__=int(query_by_procedure.query_start_time.timestamp()) * 1000
```

Comment:
You can use the method [here](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/time_utils.py#L117) to convert the timestamp to millis

Reply:
we'll update it in a separated PR